sklearn/ensemble/forest.py

```diff
@@ -153,6 +156,12 @@ def __init__(self,
         self.verbose = verbose
         self.warm_start = warm_start
         self.class_weight = class_weight
+        self.allow_missing = missing_values is not None
```

Comment:
+1, this should be determined at `fit` time, same goes for setting `self._allow_nan` below